agent/agent.go

```diff
@@ -979,8 +980,8 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 	return resp, nil
 }
 
-func (a *Agent) endTaskResponse(ctx context.Context, tc *taskContext, status string, message string) *apimodels.TaskEndDetail {
-	var userDefinedDescription string
+func (a *Agent) endTaskResponse(ctx context.Context, tc *taskContext, status string, systemFailureDescription string) *apimodels.TaskEndDetail {
+	description := systemFailureDescription
```

Comment:
Nit: can remove this line in favor of using `systemFailureDescription`

Reply:
Hmm I kind of want to leave it as a separate variable just because there's multiple failure descriptions at play, so reusing the parameter might get confusing. The priority is like this:
* User-defined task description (if any)
* System failure description (if any)
* Default failing command description (if the command failed)
* No description at all if the task succeeded _and_ would have used the default failing command description.

Reply:
Ok, that's all good with me!